spyder/plugins/maininterpreter/container.py

```diff
@@ -53,31 +53,31 @@ def update_actions(self):
 
     @on_conf_change(option=['default', 'custom_interpreter', 'custom'])
     def section_conf_update(self, option, value):
-        if option in ['default', 'custom_interpreter', 'custom'] and value:
-            self._update_status()
-            self.sig_interpreter_changed.emit()
-
-            # Set new interpreter
+        if ((option == 'default' and value) or
+                (option == 'custom' and not value)):
+            executable = get_python_executable()
+        else:
             executable = osp.normpath(self.get_conf('custom_interpreter'))
-            if (option in ['custom', 'custom_interpreter'] and
-                    osp.isfile(executable)):
+            if osp.isfile(executable):
                 self.sig_add_to_custom_interpreters_requested.emit(executable)
+            else:
+                executable = get_python_executable()
+                self.set_conf('custom_interpreter', ' ')
```

Comment:
I fully agree. Thanks!

I thought that it could have the use of also preventing programatically changes to the variable, but I hope this is not needed too.